specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/WebApps.json

```diff
@@ -22339,6 +22557,102 @@
         }
       }
     },
+    "CsmDeploymentStatus": {
+      "description": "Deployment status response payload.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "CsmDeploymentStatus resource specific properties",
+          "type": "object",
+          "properties": {
+            "deploymentId": {
+              "description": "Deployment operation id.",
+              "type": "string"
+            },
+            "status": {
+              "description": "Deployment build status.",
+              "enum": [
+                "TimedOut",
+                "RuntimeFailed",
+                "BuildAborted",
+                "BuildFailed",
+                "BuildRequestReceived",
+                "BuildPending",
+                "BuildInProgress",
+                "BuildSuccessful",
+                "PostBuildRestartRequired",
+                "StartPolling",
+                "StartPollingWithRestart",
+                "RuntimeStarting",
+                "RuntimeSuccessful"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "DeploymentBuildStatus",
+                "modelAsString": false
```

Comment:
Please be certain on this.

If you put `false`, it means you are never going to add new enum (even on new api-version).

Reply:
@JasonFreeberg I checked with Navy on this and we should set it to `true` since it is more flexible for future changes. IIRC `false` prohibits us from making changes to the enum for newer api-versions as  @weidongxu-microsoft mentioned.